src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final String SEPARATOR = ",";
+    public static final String NEW_LINE = System.lineSeparator();
+    public static final String RESULT_HEADER = "result";
+    public static final int OPERATION_INDEX = 0;
+    public static final int AMOUNT_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            String fileContent = readFileContent(fromFileName);
+            String report = formReport(fileContent);
+            writeReportToFile(toFileName, report);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading from or writing to file: "
+                    + e.getMessage(), e);
+        }
```

Comment:
You handle exceptions within private methods, no need to additionally do it in parent method
```suggestion
            String fileContent = readFileContent(fromFileName);
            String report = formReport(fileContent);
            writeReportToFile(toFileName, report);
```